### DISCUSSION TITLE
Rendering pdf images using the cairo_pdf device as default.

### DISCUSSION THREAD

**Discussion by arnaudgallou** [quarto-dev/quarto-cli#1170] [Q&A] [answered]
Is there an option to render/save figures as `pdf` using the `cairo_pdf` device as default? I've figures generated using `ggplot2` with unicode characters or transparent colors in them that don't render well with the default device. I get better results when explicitly saving figures using `ggsave` with `device = cairo_pdf`, which is not convenient.

``````
---
format: pdf
fig-format: pdf
execute: 
  echo: false
  warning: false
---

```{r}
library(tidyverse)

(ggplot() +
  geom_segment(aes(x = 0, xend = 1, y = 1, yend = 0)) +
  xlab("\u0394")) %>% 
  ggsave("test.pdf", ., device = cairo_pdf)
```

::: {#fig-a}
![](test.pdf)

This is a figure caption.
:::
``````

I’m running on macOS 12 for information.

**Comment by cderv** [comment, marked as answer]
I think you would need to use a **knitr** only solution (meaning not a Quarto configuration working also for other language). 

You can change the default device with **knitr** using the `dev` chunk option

On the chunk you need 
````r
```{r, dev='cairo_pdf'}
#| label: fig-a
#| fig-cap: This is a figure caption.
library(ggplot2)

ggplot() +
  geom_segment(aes(x = 0, xend = 1, y = 1, yend = 0)) +
  xlab("\u0394")
```
````

Or globally in a setup chunk 
````r
```{r setup, include=FALSE}
knitr::opts_chunk$set(dev = "cairo_pdf")
```
````

or using Quarto specific syntax in YAML for engine (https://quarto.org/docs/computations/execution-options.html#engine-binding)
````r
knitr:
  opts_chunk: 
    dev: cairo_pdf
````

This way, **knitr** should use `grDevices::cairo_pdf()` to save plots produced in chunks. 

Hope it helps

**Reply by arnaudgallou** [reply]
Awesome, exactly what I needed! Thank you.

**Comment by grantmcdermott** [comment]
Piggy-backing on this question: Is there a way to automatically invoke (or limit) `dev: cairo_dev` only for cases when rendering to PDF?

I normally render the same .qmd file to both HTML and PDF. Setting `knitr -> opts_chunk -> dev: cairo_pdf` globally causes my HTML rendering to fail.

(By comparison, with RMarkdown you can invoke the device directly under the document-specific YAML: `pdf(_document) -> dev: cairo_pdf` Unless, I'm missing something, this doesn't work for Quarto.)

Thanks in advance.

**Reply by baptiste** [reply]
this should do the trick,

````
```{r setup, include=FALSE}
if (knitr::is_latex_output())
  knitr::opts_chunk$set(dev = "cairo_pdf")
```
````

**Reply by grantmcdermott** [reply]
Workaround confirmed. Thanks @baptiste!
